src/ints/mouse.cpp

```diff
@@ -206,43 +179,26 @@ static void mouse_queue_tick(uint32_t)
 
 MouseQueue::MouseQueue() = default;
 
-void MouseQueue::AddEvent(MouseEvent &event)
+void MouseQueue::AddEvent(MouseEvent &ev)
 {
     DEBUG_QUEUE("AddEvent:   %s %s %s",
-                event.request_ps2 ? "PS2" : "---",
-                event.request_vmm ? "VMM" : "---",
-                event.request_dos ? "DOS" : "---");
+                ev.request_ps2 ? "PS2" : "---",
+                ev.request_vmm ? "VMM" : "---",
+                ev.request_dos ? "DOS" : "---");
 
     // From now on, do not distinguish between VMM and PS/2
     // events - virtual machine managers only
     // need to trigger PS/2 packet updates
-    event.request_ps2 |= event.request_vmm;
-
-    // If events are being fetched, clear the DOS overflow flag
-    if (mouse_shared.active_dos && !mouse_shared.dos_cb_running)
-        queue_overflow = false;
-
-    // If queue got overflowed due to DOS not taking events,
-    // don't accept any more DOS events other than mouse move,
-    // as it might lead to strange effects in DOS applications
-    if (GCC_UNLIKELY(queue_overflow) && event.request_dos &&
-        event.id != MouseEventId::MouseHasMoved) {
-        event.request_dos = false;
-        // Normal mechanism for updating button state is not working now
-        last_buttons_12S = event.buttons_12S;
-    }
-
-    // Mouse movements should be aggregated, no point in handling
-    // excessive amount of events
-    if (event.request_dos)
-        AggregateEventsDOS(event);
+    ev.request_ps2 |= ev.request_vmm;
 
-    // Prevent unnecessary further processing
-    if (!event.request_dos && !event.request_ps2)
+    // Prevent unnecessary processing
+    if (ev.request_dos)
+        AggregateDosEvents(ev);
+    if (!ev.request_dos && !ev.request_ps2)
```

Comment:
In the second `if`, the `!ev.request_dos` is the `else` portion from the first `if`.

This produces the same result, but a bit simpler?

``` c++
    if (ev.request_dos)
        AggregateDosEvents(ev);
    else if (!ev.request_ps2)
```

Reply:
Ignoring the `else` portions usually makes sense for a series of return or breaks:
``` c++
if (a)
   return a;
if (b)
   return b;
```

But without returns/breaks, we usually don't want to have to repeat the negative portion of the first if:

``` c++
if (a)
   something(a);
if (!a && b)
   something(b);
if (!a && !b && c)
   something(c);
```

becomes:
``` c++
if (a)
   something(a);
else if (b)
   something(b);
else if (c)
   something(c);
```

But If the style fits with more of the code, feel free to ignore!

Reply:
The 'AggregateDosEvents' can clean the 'ev.request_dos' flag - thus, we need both conditions in the second 'if'. In general, the event can be requested for DOS driver, for PS/2 driver, or for both, there is no rule.